python/pyspark/sql/dataframe.py

```diff
@@ -351,8 +352,62 @@ def show(self, n=20, truncate=True, vertical=False):
         else:
             print(self._jdf.showString(n, int(truncate), vertical))
 
+    def _get_repl_config(self):
+        """Return the configs for eager evaluation each time when __repr__ or
+        _repr_html_ called by user or notebook.
+        """
+        eager_eval = self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.enabled", "false").lower() == "true"
+        console_row = int(self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.maxNumRows", u"20"))
+        console_truncate = int(self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.truncate", u"20"))
+        return (eager_eval, console_row, console_truncate)
+
     def __repr__(self):
-        return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
+        (eager_eval, console_row, console_truncate) = self._get_repl_config()
+        if not self._support_repr_html and eager_eval:
+            vertical = False
+            return self._jdf.showString(
+                console_row, console_truncate, vertical)
```

Comment:
I guess

```python
return self._jdf.showString(
    console_row, console_truncate, vertical=False)
```

should work without `vertical` variable.

Reply:
Actually I firstly implement like this but we'll got an TypeError Exception
```
TypeError: __call__() got an unexpected keyword argument 'vertical'
```
The named arguments can't work during python call _jdf func.

Reply:
Oh, I see. Good to know. Thanks!